Framework/Algorithms/test/PolarizationCorrectionFredrikzeTest.h

```diff
@@ -305,20 +327,38 @@ class PolarizationCorrectionFredrikzeTest : public CxxTest::TestSuite {
                       alg->execute(), std::invalid_argument &);
   }
 
-  // Test various combinations of spin state orders
-  void test_various_spin_state_orders_for_PA() {
+  void test_valid_spin_state_order_for_PA() {
     auto groupWS = createGroupWorkspace(4, 4, 1, 1);
     auto alg = initializeAlgorithm(groupWS, "PA", "1,0,0,0");
 
```

Comment:
Hi Yusuf, just to clarify following our conversation, Adrian is suggesting here that you would have the following:

```
  void test_valid_spin_state_order_for_PA() {
    auto groupWS = createGroupWorkspace(4, 4, 1, 1);
    auto alg = initializeAlgorithm(groupWS, "PA", "1,0,0,0");
    alg->setProperty("AddSpinStateToLog", true);
```

This way we're setting the `AddSpinStateToLog` property to `true`, which this test requires, but we're doing it outside the for loop that we have later in the test. This approach should be more efficient because this property will have the same value for every iteration of the loop, so we don't need to keep setting it each time.

As we discussed, you could potentially add something to `initializeAlgorithm` to optionally set this property when required. However I think most of the tests in this class do not need to set this property, so it seems fine to leave it to be set separately. If you prefer to move it into `initializeAlgorithm`, though, then that would also be fine I think.